# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadHandler.cpp
# Commit: b8ecf23519c9
# Full Hash: b8ecf23519c9ab5180d08f00c712674bfe968dac
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 85fd9fc9cffa -r b8ecf23519c9 dom/script/ScriptLoadHandler.cpp
--- a/dom/script/ScriptLoadHandler.cpp	Mon Feb 21 15:39:08 2022 +0000
+++ b/dom/script/ScriptLoadHandler.cpp	Mon Feb 21 15:39:09 2022 +0000
@@ -11,6 +11,7 @@
 #include "ScriptLoader.h"
 #include "ScriptTrace.h"
 #include "js/Transcoding.h"
+#include "js/loader/ScriptLoadRequest.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/CheckedInt.h"
 #include "mozilla/DebugOnly.h"
@@ -24,7 +25,6 @@
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/SRICheck.h"
 #include "mozilla/dom/ScriptDecoding.h"
-#include "mozilla/dom/ScriptLoadRequest.h"
 #include "nsCOMPtr.h"
 #include "nsContentUtils.h"
 #include "nsDebug.h"
@@ -50,7 +50,7 @@
   MOZ_LOG_TEST(ScriptLoader::gScriptLoaderLog, mozilla::LogLevel::Debug)
 
 ScriptLoadHandler::ScriptLoadHandler(
-    ScriptLoader* aScriptLoader, ScriptLoadRequest* aRequest,
+    ScriptLoader* aScriptLoader, JS::loader::ScriptLoadRequest* aRequest,
     UniquePtr<SRICheckDataVerifier>&& aSRIDataVerifier)
     : mScriptLoader(aScriptLoader),
       mRequest(aRequest),
@@ -76,7 +76,7 @@
   }
 
   // Reference to the script source buffer which we will update.
-  ScriptLoadRequest::ScriptTextBuffer<Unit>& scriptText =
+  JS::loader::ScriptLoadRequest::ScriptTextBuffer<Unit>& scriptText =
       mRequest->ScriptText<Unit>();
 
   uint32_t haveRead = scriptText.length();